src/mobile/src/ui/views/wallet/Loading.js

```diff
@@ -163,19 +169,35 @@ class Loading extends Component {
 
         if (addingAdditionalAccount) {
             const seedStore = new SeedStore[additionalAccountType](password, additionalAccountName);
-            this.props.getFullAccountInfo(seedStore, additionalAccountName, navigator);
+            this.props.getFullAccountInfo(seedStore, additionalAccountName);
         } else {
             const seedStore = new SeedStore[selectedAccountType](password, selectedAccountName);
-            this.props.getAccountInfo(seedStore, selectedAccountName, navigator);
+            this.props.getAccountInfo(seedStore, selectedAccountName);
         }
     }
 
     componentWillReceiveProps(newProps) {
-        const { ready, addingAdditionalAccount } = this.props;
+        const {
+            ready,
+            addingAdditionalAccount,
+            hasErrorFetchingAccountInfo,
+            hasErrorFetchingFullAccountInfo,
+            accountNames,
+        } = this.props;
         const isReady = !ready && newProps.ready;
         if ((isReady && this.state.animationPartOneDone) || (isReady && addingAdditionalAccount)) {
             this.launchHomeScreen();
         }
+        if (!hasErrorFetchingAccountInfo && newProps.hasErrorFetchingAccountInfo) {
+            this.redirectToLogin();
+        }
+        if (!hasErrorFetchingFullAccountInfo && newProps.hasErrorFetchingFullAccountInfo) {
+            if (accountNames.length <= 1) {
```

Comment:
What is the reason we'll directly navigate to dashboard (if there are multiple accounts)?

Reply:
For when you're adding an additional account